### PR TITLE
Use Consumer.Close in code defensively

### PR DESCRIPTION
**Pull request by Arkatufus**

### CODE REVIEW

File: src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs [hunk 1]
```diff
@@ -259,24 +259,36 @@ private void ApplySettings(ConsumerSettings<K, V> updatedSettings)
 
         protected override void PostStop()
         {
-            if (_settings.ConnectionCheckerSettings.Enabled)
+            base.PostStop();
```
Comment (IgorFedchenko) [inline src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs:262]: So in alpakka base `PostStop` is executed before the rest of the handling?

**Review by IgorFedchenko — COMMENTED** [review]
Left some questions about changes in `KafkaConsumerActor.PostStop` logic, which are not related to using `Close` method. 

As you mentioned before - the more patches we apply, the harder to keep stuff in sync with scala version. Does alpakka has this `PostStop` method implemented like this? Because to me, there is no reason to update this method for now, unless it does not match alpakka implementation (at the moment of porting, it was matching I think).

**Review by IgorFedchenko — APPROVED** [review]
Let's put it like this @Arkatufus : if you think that this change gives more benefits then takes (I mean the difference with alpakka code) - merge it, since as is it looks good to me, and passes all the tests.

**Comment by Aaronontheweb** [comment]
@IgorFedchenko @Arkatufus have conflicts here

**Pull request merged**